### PR TITLE
add distance limit in error message for 154 error code

### LINKED ISSUE #3722: Show distance limit in error message
When I attempt to plan a longer route at https://valhalla.openstreetmap.de/ I get the error message:

> OSM: Path distance exceeds the max distance limit for bicycle.

It would be nice if the error message said what the distance limit is.

### PR DESCRIPTION
**Pull request by nilsnolde**
fixes #3722 

@jackrea07 & @reillywynn , in case you want to have a look. Since it's really annoying for the web app to not display this info, I did it quickly myself.

@kevinkreiser, regarding your comment here: https://github.com/valhalla/valhalla/pull/3752#issuecomment-1250877235

In fact, most other errors are already returning the server-side limit. Maybe it'd make sense to return both, the actual limit and the passed value, since the passed value is often not readily comprehensible (e.g. crow distance etc). I went without for now, since most existing errors also simply state the server limit.

### CODE REVIEW

File: src/loki/polygon_search.cc [hunk 1]
```diff
@@ -109,7 +109,7 @@ edges_in_rings(const google::protobuf::RepeatedPtrField<valhalla::Ring>& rings_p
     rings_length += bg::perimeter(ring_bg, Haversine());
   }
   if (rings_length > max_length) {
-    throw valhalla_exception_t(167, std::to_string(max_length));
```
Comment (nilsnolde) [inline src/loki/polygon_search.cc:112]: meter precision is enough IMO:)

File: src/loki/matrix_action.cc [hunk 2]
```diff
@@ -35,7 +35,8 @@ void check_distance(const google::protobuf::RepeatedPtrField<valhalla::Location>
       }
 
       if (path_distance > matrix_max_distance) {
-        throw valhalla_exception_t{154};
+        throw valhalla_exception_t{154, std::to_string(static_cast<size_t>(matrix_max_distance)) +
```
Comment (nilsnolde) [inline src/loki/matrix_action.cc:38]: is there somewhere a function already that could replace `std::to_string(static_cast<size_t>())`? if not, would it make sense to add one to `midgard.util.h` or so?

File: src/worker.cc [hunk 3]
```diff
@@ -1073,7 +1073,7 @@ valhalla_exception_t::valhalla_exception_t(unsigned code, const std::string& ext
     *this = code_itr->second;
   }
   if (!extra.empty())
-    message += ":" + extra;
+    message += ": " + extra;
```
Comment (nilsnolde) [inline src/worker.cc:1076]: IMO it's nicer with a space after a colon

**Review by kevinkreiser — APPROVED** [review]

**Comment by nilsnolde** [comment]
can you re-approve @kevinkreiser ?

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**